src/qudi/interface/scannable_laser_interface.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+"""
+Interface file for lasers whose frequency can be scanned and also stabilized
+
+Copyright (c) 2021, the qudi developers. See the AUTHORS.md file at the top-level directory of this
+distribution and on <https://github.com/Ulm-IQO/qudi-iqo-modules/>
+
+This file is part of qudi.
+
+Qudi is free software: you can redistribute it and/or modify it under the terms of
+the GNU Lesser General Public License as published by the Free Software Foundation,
+either version 3 of the License, or (at your option) any later version.
+
+Qudi is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along with qudi.
+If not, see <https://www.gnu.org/licenses/>.
+"""
+
+__all__ = ('ScannableLaserInterface', 'ScannableLaserConstraints', 'LaserScanMode',
+           'LaserScanDirection', 'ScannableLaserSettings', 'StabilizationMixin')
+
+from abc import abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Union, Iterable, Optional, Tuple
+
+from qudi.core.module import Base
+from qudi.util.constraints import ScalarConstraint
+
+from qudi.interface.process_control_interface import ProcessControlConstraints, ProcessSetpointInterface, \
+    ProcessControlInterface
+
+
+class LaserScanMode(Enum):
+    CONTINUOUS = 0
+    FIXED_REPETITIONS = 1  # Includes single sweep
+
+
+class LaserScanDirection(Enum):
+    UP = 0
+    DOWN = 1
+
+
+@dataclass(frozen=True)
+class ScannableLaserConstraints:
+
+    scan_limits: Tuple[float, float]
```

Comment:
Why not use also `ScalarConstraint` here? This would make things more consistent and it comes with built-in checker methods.

Reply:
Was originally intending to do so, but then got a bit confused on using two scalar constraints for such a thing with upper and lower limits.

Just went for two floats out of convenience, because I thought its easier for such a simple concept.

Reply:
We would only need a single scalar constraint here though, right? I understand the scan limits as the minimum and maximum value that the scan "position" in terms of scan units can reach. The constraint then includes the two floats for the upper and lower bound.